### ISSUE TITLE
Issues related to Delete image(s) and Form upload image which are loaded on init

### ISSUE THREAD

**Issue by NSAKHAN** [CreativeDream/jquery.filer#26]
Hello there,

First i like to appreciate about this module.

I found some issues related to deleting an image and uploading when an image is loaded in files using jquery filer init. 

On page load, i'm loading images URL and initialising it to jquery.filer files array. It's showing up perfectly. 
When i'm deleting any of those images it's deleting perfectly from front-end and back-end (fixed in 1.0.3). 
But when i'm adding more images after selecting it from filesystem, the old images are not been deleted, only the newly added images are deleting. Secondly it works fine if I delete all the newly added images then delete the old i.e. loaded images from files array, then it got deleted.

Second issue i found regarding form upload. Or might be i'm unaware of it how to handle it. When i'm loading the images through jquery.filer while init by specifying the URL's. It's not sending those image files to server on submit form.

Kindly let me know if you didn't understand. I will explain it again.

Thanks

**Comment by nirajan-panthee** [comment]
I got the similar issue When i'm loading the images through jquery.filer while init by specifying the URL's. It's not sending those image files to server on submit form.

**Comment by CreativeDream** [comment]
First issue: I can't reproduce it :(
Second issue: You want to have on back-end a list with all files that are already uploaded(they were loaded to jquery.filer in option: "files") ?

**Comment by CreativeDream** [comment]
@NSAKHAN 
First issue: 
I can't reproduce it.

Second issue: 
You want to have on back-end a list with all files that are already uploaded(they were loaded to jquery.filer in option: "files") ?

**Comment by NSAKHAN** [comment]
Firstly let me tell you about my requirement: 

 My requirement is I have a New Form. I want to add images to it and submit the form. My Api stores the data as well as the images. Now in my edit form. I'm displaying the uploaded images and if a user want to append more images to it then he may append more and submit the form. Now on the server side request should contain all the images i.e. old images which were already uploaded and the newly added images as well. Currently i'm getting only newly added images by jquery-filer.

Let me explain briefly in steps with Issue and Fine Tags:

Overview:

1. I have a form with one jquery.filer Input Tag (multiple images upload). [Fine]
2. I'm only doing api call on form submission. Nothing is updating , adding or deleting if a user add/remove images. [Fine]
3. Pictures are adding and deleting from form from front-end as well as back-end [though this was not solved on previous release, glad u fixed it] 
 (NOTE: before submission, when user only plays arounds, i.e. adding deleting images just on form).[Fine]
4. After submission, my api stores the images on the server filesystem. Working fine till here. [Fine]

[ISSUE]

5. Now on the same form (in Edit Mode) i'm getting the URL's of the images which i recently stored and adding it into jquery.filer files option. Following is the code.

var images = // this is an array of images im populating, not included code due to brevity
 $('#fileUpload').filer({
            changeInput: '<div class="jFiler-input-dragDrop"><div class="jFiler-input-inner"><div class="jFiler-input-icon"><i class="icon-jfi-cloud-up-o"></i></div><div class="jFiler-input-text"><h3>Drag&Drop files here</h3> <span style="display:inline-block; margin: 15px 0">or</span></div><a class="jFiler-input-choose-btn blue">Browse Files</a></div></div>',
            showThumbs: true,
            theme: "dragdropbox",
            templates: {
                box: '<ul class="jFiler-item-list"></ul>',
                item: '<li class="jFiler-item">\
                            <div class="jFiler-item-container">\
                                <div class="jFiler-item-inner">\
                                    <div class="jFiler-item-thumb">\
                                        <div class="jFiler-item-status"></div>\
                                        <div class="jFiler-item-info">\
                                            <span class="jFiler-item-title"><b title="@@fi-name@@">@@fi-name | limitTo: 25@@</b></span>\
                                        </div>\
                                        @@fi-image@@\
                                    </div>\
                                    <div class="jFiler-item-assets jFiler-row">\
                                        <ul class="list-inline pull-left">\
                                            <li><span class="jFiler-item-others">@@fi-icon@@ @@fi-size2@@</span></li>\
                                        </ul>\
                                        <ul class="list-inline pull-right">\
                                            <li><a class="icon-jfi-trash jFiler-item-trash-action"></a></li>\
                                        </ul>\
                                    </div>\
                                </div>\
                            </div>\
                        </li>',
                itemAppend: '<li class="jFiler-item">\
                            <div class="jFiler-item-container">\
                                <div class="jFiler-item-inner">\
                                    <div class="jFiler-item-thumb">\
                                        <div class="jFiler-item-status"></div>\
                                        <div class="jFiler-item-info">\
                                            <span class="jFiler-item-title"><b title="@@fi-name@@">@@fi-name | limitTo: 25@@</b></span>\
                                        </div>\
                                        @@fi-image@@\
                                    </div>\
                                    <div class="jFiler-item-assets jFiler-row">\
                                        <ul class="list-inline pull-left">\
                                            <span class="jFiler-item-others">@@fi-icon@@ @@fi-size2@@</span>\
                                        </ul>\
                                        <ul class="list-inline pull-right">\
                                            <li><a class="icon-jfi-trash jFiler-item-trash-action"></a></li>\
                                        </ul>\
                                    </div>\
                                </div>\
                            </div>\
                        </li>',
                progressBar: '<div class="bar"></div>',
                itemAppendToEnd: true,
                dragDrop: {
                    dragEnter: function(){},
                    dragLeave: function(){},
                    drop: function(){} 
                },
                clipBoardPaste:true,
                removeConfirmation: false,
                _selectors: {
                    list: '.jFiler-item-list',
                    item: '.jFiler-item',
                    progressBar: '.bar',
                    remove: '.jFiler-item-trash-action'
                }
            },
            addMore: true, 
            files: images
        });

Second Issue:

6. Images are showing up perfectly. But the images which i added by URL did not get added to form input. When i add more images to it (i.e. appending more images to forms) and submit the form. Only newly added images uploads. Not the images in files array.


First Issue:

7.The time when form is loaded with images. Delete button is working fine. But when  i append more images to it. Delete button doesn't work for previous images. Let me explain with example.

Example : 

I have total 3 images i.e. abc.png, abc1.png, abc2.png

I already uploaded these images, and now im on form and i can see the images coming from files array.

Now i added 2 more images i.e. abc3.png, abc4.png,  it shows added up on form fine.

Now if i click on the delete button of any one of abc.png, abc1.png, abc2.png, delete button does not remove the image from the front-end but add the image details in the image-removed array of the form.

**Comment by ekosijabat** [comment]
Hello there,

First thanks a lot for this plugin. I like this plugin because it's user friendly.

But i've got same issue like NSAKHAN. And I've tried to debug every variable at your filer.js.

- I tried console.log(s) at _changeInput function, and it only return 1 object.
And the content of the object only like this : 
```
Object[input property value = "" attribute value = "null"].
```

- I click the object and it show me this part :

```
<input type="file" name="files[]" style="position: absolute; left: -9999px; top: -9999px; z-index: -9999;" multiple="multiple">
```

In NSAKHAN's case, because he has 3 images, i think this should return 3 object. 

The next issue is still from the above condition, there is no value inside property of the object. I think it should return the name of the file (abc.png, or abc1.png or abc2.png).

I said that because when i tried to upload an image, then i deleted it, it return me object like this : 
```
Object[input property value = "191a1a.jpg" attribute value = "null"]
```
This object return when i debug console.log(current_input) at _remove function


I'm so sorry that i can't help much to fix this bug. I don't have experience using javascript and jQuery.
